TombEngine/Scripting/Internal/TEN/Inventory/InventoryHandler.cpp

```diff
@@ -21,26 +24,37 @@ namespace TEN::Scripting::InventoryHandler
 	/// Add an item to the player's inventory.
 	//@function GiveItem
 	//@tparam Objects.ObjID objectID Object ID of the item to add.
-	//@int[opt] count The amount of items to add. Default is the yield from a single pickup, e.g. 1 from a medipack, 12 from a flare pack.
-	//@bool[opt] addToPickupSummary If true, display the item in the pickup summary. Default is false.
-	static void GiveItem(GAME_OBJECT_ID objectID, sol::optional<int> count, sol::optional<bool> addToPickupSummary)
+	//@tparam[opt=1] int count The amount of items to add. Default is the yield from a single pickup, e.g. 1 from a medipack, 12 from a flare pack.
+	//@tparam[opt=false] bool addToPickupSummary If true, display the item in the pickup summary. Default is false.
+	static void GiveItem(GAME_OBJECT_ID objectID, TypeOrNil<int> count, TypeOrNil<bool> addToPickupSummary)
 	{
-		PickedUpObject(objectID, count.has_value() ? std::optional<int>(*count) : std::nullopt);
+		auto convertedCount = ValueOr<int>(count, 1);
+
+		PickedUpObject(objectID, convertedCount);
 		
-		if (addToPickupSummary.value_or(false))
+		if (ValueOr<bool>(addToPickupSummary, false))
 		{
-			auto pos = GetJointPosition(LaraItem, LM_HIPS).ToVector3();
-			g_Hud.PickupSummary.AddDisplayPickup(objectID, pos, count.value_or(1));
+			auto startPosition = Vector2(110.0f, 85.0f);
+			
+			// Convert percentage-based position (0–100%) to screen pixel coordinates
+			float fWidth = g_Configuration.ScreenWidth;
```

Comment:
Pickup position calculation is incorrect, and results are different on different resolutions.